src/crimson/os/seastore/seastore.cc

```diff
@@ -1265,7 +1266,44 @@ SeaStore::tm_ret SeaStore::_remove(
 {
   LOG_PREFIX(SeaStore::_remove);
   DEBUGT("onode={}", *ctx.transaction, *onode);
-  return onode_manager->erase_onode(*ctx.transaction, onode);
+  auto fut = BtreeOMapManager::omap_clear_iertr::now();
+  auto omap_root = onode->get_layout().omap_root.get(
+    onode->get_metadata_hint(device->get_block_size()));
+  if (omap_root.get_location() != L_ADDR_NULL) {
+    fut = seastar::do_with(
+      BtreeOMapManager(*transaction_manager),
+      onode->get_layout().omap_root.get(
+	onode->get_metadata_hint(device->get_block_size())),
+      [&ctx, onode](auto &omap_manager, auto &omap_root) {
+      return omap_manager.omap_clear(
+	omap_root,
+	*ctx.transaction
+      );
+    });
+  }
+  return fut.si_then([this, &ctx, onode] {
+    auto object_data =
+      onode->get_layout().object_data.get();
+    return transaction_manager->dec_ref(
+      *ctx.transaction,
+      object_data.get_reserved_data_base(),
+      object_data.get_reserved_data_len()
+    ).si_then([](auto refcounts) {
```

Comment:
The architecture here seems to be SeaStore::_remove -> TransactionManager::dec_ref .  I don't actually see a user for LBAManager::reclaim_space.  Also, rather than doing this directly, SeaStore::_remove should probably be invoking ObjectDataManager::clear.

Reply:
All done, please take a look again